### PR TITLE
fix(mj + slack): log exception, guess at fixing

### PR DESCRIPTION
**Pull request by emjburns**
This week I added a feature to hide long commits in a pop up. Apparently, when we update the manual judgment notification we scrape the text from the original message and re-add it to the notification with the updated image and title. I broke that.

This PR takes a stab at fixing it, and also logs the exception we get so I can try and really fix it.

### CODE REVIEW

File: keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt [hunk 1]
```diff
@@ -63,13 +64,25 @@ class ManualJudgmentCallbackHandler(
    * For example, if user gyardeni approved the notification, this function will add:
    * "@Gal Yardeni hit approve on 2021-02-12 11:05:57 AM" and will marked the original text with strikethrough.
    */
-  fun updateManualJudgementNotification(response: BlockActionPayload): List<LayoutBlock> {
+  fun oldUpdateManualJudgementNotification(response: BlockActionPayload): List<LayoutBlock> {
```
Comment (luispollo) [inline keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt:67]: Why the rename?

Comment (emjburns) [reply keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt:67]: arg oops, good catch. it was an accident from my other idea. I'll fix

File: keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt [hunk 2]
```diff
@@ -63,13 +64,25 @@ class ManualJudgmentCallbackHandler(
...
     try {
       val originalCommitText = response.message.blocks[1].getText
-      val originalGitInfoText = response.message.blocks[2].getText
-      val originalUrl = response.message.blocks[2].getUrl
       val action = actionsMap[response.actions.first().value]
 
+      // since we sometimes add a "show more" button, the position of this information changes.
+      //  I cannot figure out (yet) how to scrape the original section and just re-add it to the notification
```
Comment (luispollo) [inline keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt:73]: Aren't the blocks all exactly the same except for the one added at the bottom to say "X person approved/rejected"? If that's the case, couldn't we just use the original array of blocks and tack that at the end? Why do we need to worry about these positions to begin with?

Comment (emjburns) [reply keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt:73]: We update the first block to have a different image.

I have been struggling with how to just grab the old blocks but actually, you just gave me an idea!

File: keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt [hunk 3]
```diff
@@ -104,8 +90,13 @@ class ManualJudgmentCallbackHandler(
         }
       }
 
+      val originalBlocks = response.message.blocks
+      //remove the first two blocks because we're replacing them
+      originalBlocks.removeFirstOrNull()
+      originalBlocks.removeFirstOrNull()
+      return newBlocks + originalBlocks
```
Comment (luispollo) [inline keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt:97]: Perfect! Something like that is what I had in mind!

**Review by luispollo — APPROVED** [review]

File: keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt (hunk 3 above)
Comment (emjburns) [reply keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt:97]: I was so stuck on working inside the `withBlocks` thing that I forgot it returned a list of blocks. 🤞

**Pull request merged**